packages/components/src/popover/index.js

```diff
@@ -587,7 +595,7 @@ const Popover = (
 				}
 			) }
 			{ ...contentProps }
-			onKeyDown={ maybeClose }
+			onKeyDown={ onKeyDown }
```

Comment:
Nit pick: you could remove `onKeyDown` from the props and then allow it to be spread through `...contentProps` in the line above.